eZ/Bundle/EzPublishIOBundle/Command/MigrateFilesCommand.php

```diff
@@ -0,0 +1,276 @@
+            ->addOption('bulk-count', null, InputOption::VALUE_REQUIRED, 'Number of files processed at once', 100)
+            ->addOption('dry-run', null, InputOption::VALUE_NONE, 'Execute a dry run')
+            ->setHelp(
+                <<<EOT
+The command <info>%command.name%</info> migrates files from one IO repository
+to another.
+
+It can for example be used to migrate local files from the default IO
+configuration to a new IO configuration, like a clustered setup.
+
+The <info>--from</info> and <info>--to</info> values must be specified as <info><metadata_handler>,<binarydata_handler></info>.
+If <info>--from</info> is omitted, the default IO configuration will be used.
+If <info>--to</info> is omitted, the first non-default IO configuration will be used.
+
+<fg=red>During the script execution the files should not be modified. To avoid
+surprises you are advised to create a backup and/or execute a dry run before
+proceeding with actual update.</>
+
+Since this script can potentially run for a very long time, to avoid memory
+exhaustion run it in production environment using <info>--env=prod</info> switch.
+
+EOT
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if ($input->getOption('list-io-configs')) {
+            $this->outputConfiguredHandlers($output);
+
+            return;
+        }
+
+        $output->writeln($this->getProcessedHelp());
+
+        $fromHandlers = $input->getOption('from') ? explode(',', $input->getOption('from')) : null;
+        $toHandlers = $input->getOption('to') ? explode(',', $input->getOption('to')) : null;
+        if (!$this->areHandlerOptionsValid($fromHandlers, $toHandlers, $output)) {
+            return;
+        }
+
+        if (!$fromHandlers) {
+            $fromHandlers = ['default', 'default'];
+        }
+        if (!$toHandlers) {
+            $toHandlers = [
+                array_keys($this->configuredMetadataHandlers)[0],
+                array_keys($this->configuredBinarydataHandlers)[0],
+            ];
+        }
+
+        $output->writeln([
+            "Migrating from '$fromHandlers[0],$fromHandlers[1]' to '$toHandlers[0],$toHandlers[1]'",
+            '',
+        ]);
+        $this->migrationHandler->setIODataHandlersByIdentifiers(
+            $fromHandlers[0],
+            $fromHandlers[1],
+            $toHandlers[0],
+            $toHandlers[1]
+        );
+
+        $totalCount = $this->migrationHandler->countFiles();
+        $output->writeln([
+            'Found total files to update: ' . $totalCount,
+            '',
+        ]);
+
+        if ($totalCount === 0) {
+            $output->writeln('Nothing to process.');
+
+            return;
+        }
+
+        $helper = $this->getHelper('question');
+        $question = new ConfirmationQuestion(
+            '<question>Are you sure you want to proceed?</question> ',
```

Comment:
No, because I have do lots of rewriting then :) AFAIK we have never translated console scripts before. We can change that of course, but it's indeed a wide topic.